sklego/model_selection.py

```diff
@@ -244,7 +267,382 @@ def split(self, X, y=None, groups=None):
 
         for label in np.unique(clusters):
-            yield (np.where(clusters != label)[0], np.where(clusters == label)[0])
+            yield (
+                np.where(clusters != label)[0],
+                np.where(clusters == label)[0],
+            )
+
+    def _method_is_fitted(self, X):
+        """
+        :param X: Array to use if the method is fitted
+        :return: True if fitted, else False
+        """
+        try:
+            self.cluster_method.predict(X[0:1, :])
+            return True
+        except NotFittedError:
+            return False
+
+
+class GroupTimeSeriesSplit(_BaseKFold):
+    """
+    Sliding window time series split
+
+    - Create n_splits folds with an as equally possible size through a smart variant of a brute
+    force search. Groups parameter in .split() should be filled with the time groups (e.g. years)
+
+    :param n_splits: the amount of train-test combinations.
+    :type n_splits: int
+
+    with n_splits at 3
+    * = train
+    x = test
+    |-----------------------|
+    | * * * x x x - - - - - |
+    | - - - * * * x x x - - |
+    | - - - - - - * * * x x |
+    |-----------------------|
+    """
+    # table above inspired by sktime
+
+    def __init__(self, n_splits):
+        if not isinstance(n_splits, numbers.Integral):
+            raise ValueError(
+                "The number of folds must be of Integral type. "
+                "%s of type %s was passed." % (n_splits, type(n_splits))
+            )
+        n_splits = int(n_splits)
+
+        if n_splits <= 1:
+            raise ValueError(
+                "k-fold cross-validation requires at least one"
+                " train/test split by setting n_splits=2 or more,"
+                " got n_splits={0}.".format(n_splits)
+            )
+
+        self.n_splits = n_splits
+
+    def summary(self):
+        """
+        Generates a pd.DataFrame which displays the groups splits
+        and extra statistics about it.
+        Can only be run after having applied the .split() method to
+        the GroupTimeSeriesSplit instance.
+
+        :return: a pd.DataFrame with info about where the splits were made
+        :rtype: pd.DataFrame
+        """
+        try:
+            return (
+                self._grouped_df.sort_index()
+                .assign(group=lambda df: df["group"].astype(int))
+                .assign(
+                    obs_per_group=lambda df: df.groupby("group")[
+                        "observations"
+                    ].transform("sum")
+                )
+                .assign(ideal_group_size=round(self._ideal_group_size))
+                .assign(
+                    diff_from_ideal_group_size=lambda df: df["obs_per_group"]
+                    - df["ideal_group_size"]
+                )
+            )
+        except AttributeError:
+            raise AttributeError(
+                ".summary() only works after having ran"
+                " .split(X, y, groups)."
+            )
```

Comment:
I wouldn't mind having such a utility function in the documentation, but does it have to be a part of the public API? Do other sklearn components implement this?

Reply:
`TimeGapSplit` of scikit-lego also has a method called `.summary()` which returns a `pd.DataFrame`, which I used as inspiration for this method.  I do believe this method is very helpful and informative for users to help understand the final group division.